devito/operator.py

```diff
@@ -190,14 +190,23 @@ def arguments(self, *args, **kwargs):
         # Insert loop size arguments from dimension values
         d_args = [d for d in arguments.values() if isinstance(d, Dimension)]
         for d in d_args:
-            arguments[d.name] = dim_sizes[d]
+            if d not in dim_sizes and d.is_Buffered:
+                # Only the parent is defined, so use it
+                arguments[d.name] = dim_sizes[d.parent]
+            else:
+                arguments[d.name] = dim_sizes[d]
```

Comment:
if ``d`` were not in ``dim_sizes``, then it must be ``d.is_Buffered``, because otherwise ``dim_sizes[d]`` (the ``else`` branch) would fail, right? So perhaps what I wrote is wrong, but there's something weird going on in the ``if`` condition?